components/AssetsList/BaseGovernanceForm-data.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  GovernanceConfig,
+  VoteThresholdType,
+  VoteTipping,
+} from '@solana/spl-governance'
+import { DISABLED_VOTER_WEIGHT } from '@tools/constants'
+import { isDisabledVoterWeight } from '@tools/governance/units'
+import {
+  getTimestampFromDays,
+  getDaysFromTimestamp,
+  parseMintNaturalAmountFromDecimalAsBN,
+  fmtBnMintDecimalsUndelimited,
+} from '@tools/sdk/units'
+
+type FormErrors<T> = {
+  [P in keyof T]: Error | undefined
+}
+
+/* all token values in human-readable units */
+export type BaseGovernanceFormFieldsV3 = {
+  /* in days */
+  minInstructionHoldUpTime: string
+  /* in days */
+  maxVotingTime: string
+
+  // 'disabled' for disabled values
+  minCommunityTokensToCreateProposal: string | 'disabled'
+  // 'disabled' for disabled values
+  minCouncilTokensToCreateProposal: string | 'disabled'
+
+  communityVoteThreshold: string | 'disabled'
+  communityVetoVoteThreshold: string | 'disabled'
+  councilVoteThreshold: string | 'disabled'
+  councilVetoVoteThreshold: string | 'disabled'
+  communityVoteTipping: VoteTipping
+  councilVoteTipping: VoteTipping
+  _programVersion: 3
+}
+
+type Transformer<T extends Record<keyof U, any>, U> = {
```

Comment:
For the base governance rules proposal form, instead of using something like Yup or Zod to parse the form and convert it into config, I just wrote a super bare bones parser framework. I think in the future I would use Zod but I thought this worked well enough.